### PR TITLE
Reset performance baseline to accommodate changes done when replacing compression libraries 

### PR DESCRIPTION
**Pull request by jbartok**
Accept performance regression introduced in https://github.com/gradle/gradle/pull/19570 . Reasons behind the decision:
* There is one ~4% performance regression on visiting TAR trees. 
* The is one ~12% performance improvement on visiting ZIP trees. 
* Both are visible on very synthetic test scenarios. 
* The impact on real world scenarios is unclear but the results mostly show small improvements here and there.
* The somewhat faster handling of ZIP archives might help in more realistic scenarios.

There is also another performance test scenario which failed when making this change: assemble for abi change | largeMonolithicJavaProject | JavaIncrementalExecutionPerformanceTest. We have looked at that and seem like it was already on the verge of failing (for a while), plus it's about a single project build, so we decided to ignore it.

Re-enable performance tests disabled in the above-mentioned PR.

Reset the performance baseline to the merge commit of the above-mentioned PR.

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle test APT.

**Review by eskatos — APPROVED** [review]
👍

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/java/JavaIncrementalExecutionPerformanceTest.groovy [hunk 1]
```diff
@@ -49,7 +49,7 @@ class JavaIncrementalExecutionPerformanceTest extends AbstractIncrementalExecuti
     }
 
     @RunFor([
-        @Scenario(type = PER_COMMIT, operatingSystems = LINUX, testProjects = ["largeGroovyMultiProject", "largeMonolithicGroovyProject"], iterationMatcher = "assemble for non-abi change"),
+        @Scenario(type = PER_COMMIT, operatingSystems = LINUX, testProjects = ["largeGroovyMultiProject", "largeMonolithicJavaProject", "largeMonolithicGroovyProject"], iterationMatcher = "assemble for non-abi change"),
```
Comment (wolfs) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/java/JavaIncrementalExecutionPerformanceTest.groovy:52]: Is there a regression in `largeMonolithicJavaProject` as well? Can you add this to the PR description?

Comment (jbartok) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/java/JavaIncrementalExecutionPerformanceTest.groovy:52]: Done. Also, for full tracking pls. see https://github.com/gradle/gradle-private/issues/3623.

**Comment by jbartok** [comment]
@bot-gradle test APT

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Pull request merged**